app/controllers/submissions_controller.rb

```diff
@@ -410,6 +410,13 @@ def update_files
             authorize! to: :manage_subdirectories? # ensure user is authorized for directories in zip files
             success, msgs = add_folder(f, current_role, repo, path: path, txn: txn, required_files: required_files)
           else
+            content_type = Marcel::MimeType.for Pathname.new(f)
+            file_extension = File.extname(f.original_filename).downcase
+            expected_mime_type = Marcel::MimeType.for extension: file_extension
+
+            if content_type != expected_mime_type && content_type != 'application/octet-stream'
```

Comment:
Okay I looked into this a bit more. This is indeed pretty tricky. For this PR it's okay to leave the behaviour the way it is, where this doesn't give a (back-end) warning if renaming a `.txt` to a `.pdf`. Doing anything else will be trickier, and it's better in general to have false negatives (no message shown) rather than false positives (message shown but renaming is actually okay).